client/my-sites/checkout/composite-checkout/hooks/use-shopping-cart-manager/use-initialize-cart-from-server.ts

```diff
@@ -43,10 +46,32 @@ export default function useInitializeCartFromServer(
 
 		if ( isInitialized.current ) {
 			debug( 'not initializing cart; already initialized' );
+			if ( productsToAddOnInitialize !== initializedProducts.current ) {
```

Comment:
This is always returning true (and therefore always logging the console warning for me), even when the cart is working correctly.  I think it's because the objects are never the same actual object instance?  Using `isEqual()` here instead might be more like what we want, if the goal is just to make sure the list of products is the same.

Reply:
😢 That's annoying. I know it's checking for referential equality, but the objects should be the same objects because they're being returned by a hook that should be memoizing its results. I'll take a closer look.

Reply:
I was not able to reproduce this issue except for the real warnings that were being caused by the bug we're trying to fix. Are you sure you see this all the time? If so, can you copy/paste an example of the warning where the two arrays contain the same data?

Reply:
Nevermind, I was able to reproduce it.